test/Controller/ArticleControllerTest.php

```diff
@@ -1171,7 +1171,7 @@ public function it_displays_content()
         $this->assertSame('10.7554/eLife.00001', $crawler->filter('meta[name="citation_doi"]')->attr('content'));
         $this->assertSame('eLife 2007;1:e00001', $crawler->filter('meta[name="citation_id"]')->attr('content'));
         $this->assertSame('http://www.example.com/pdf', $crawler->filter('meta[name="citation_pdf_url"]')->attr('content'));
-        $this->assertSame('http://www.example.com/xml', $crawler->filter('meta[name="citation_xml_url"]')->attr('content'));
+        $this->assertSame('http://localhost/articles/00001.xml', $crawler->filter('meta[name="citation_xml_url"]')->attr('content'));
```

Comment:
Understanding that the hostname changes because it's now journal's rather than the origin of the XML (some CDN/bucket)

Reply:
This example is version-less, is the version-ful XML URL used anywhere? It's in `routing.yml` so it should be tested if it's provided

Reply:
While this tests the link, the two new routes need a controller test as well (probably borrowing the mock for the streaming request from the existing one for `DownloadController`)

There is also the option of reusing `DownloadController` by redirecting to it from `xmlAction`. That would leave us with easy-to-write links but avoid testing streaming twice, thoughts?

Reply:
`DownloadController` can be reused (using an internal redirect), adds the `Content-Disposition` header (but should be harmless, and could be unset).

Happy with this approach though, but most of the contents of `DownloadControllerTest` would need to be in dedicated unit tests of the new service.